### ISSUE TITLE
[surfman] Unexpected WebGL error: 0x502 (1282) [BindVertexArray(None)]'

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#24593]
When running BabylonJS demos (like [this one](http://paulrouget.com/test/bbjs/basic/) on [servo+surfman](https://github.com/asajeffrey/servo/tree/surfman) on UWP, I get the following error in 2D mode:

```
WebGL thread' panicked at 'assertion failed: `(left == right)`
  left: `1282`,
 right: `0`: Unexpected WebGL error: 0x502 (1282) [BindVertexArray(None)]', components\canvas\webgl_thread.rs:1749:9
```

cc @asajeffrey @jdm

**Comment by jdm** [comment]
I think there are a couple issues at play here, but fundamentally it seems like it's our mistake. We enable the OES_vertex_array_object webgl extension based on the GL_OES_vertex_array_object/GL_ARB_vertex_array_object extensions reported by the driver. These extensions provide GenVertexArraysOES/BindVertexArraysOES/IsVertexArraysOES/DeleteVertexArraysOES functions, but we're not using those in sparkle: https://github.com/servo/sparkle/blob/94325e87bc18c697045be5b17ceb37f04acd9431/src/lib.rs#L78-L86

We're generating bindings for ES3 which include these functions (https://github.com/servo/sparkle/blob/94325e87bc18c697045be5b17ceb37f04acd9431/build.rs#L38), but requesting ES2 contexts which don't support them.

**Comment by jdm** [comment]
Fixed as part of #24482.
